repo_autoindex/_impl/kickstart.py

```diff
@@ -133,14 +133,17 @@ async def _treeinfo_entries(self) -> list[IndexEntry]:
         treeinfo = configparser.ConfigParser()
         treeinfo.read_string(self.treeinfo_content)
 
-        for image in treeinfo["checksums"]:
-            entry = IndexEntry(
-                href=image,
-                text=os.path.basename(image),
-            )
-            if entry.href.endswith(".iso") or entry.href.endswith(".img"):
-                entry.icon = ICON_OPTICAL
-            out.append(entry)
+        # AppStream kickstart repos uniquely do not contain images, and as such, do not
+        # contain a 'checksums' section.
+        if not treeinfo.get("general", "variant") == "AppStream":
```

Comment:
I don't like that this is hardcoding a solution just for one particular convention followed in one Red Hat variant shipped today.

If the file format itself allows this section to be absent, would it be better to design around that?  I mean like this:

```python
for image in treeinfo.get("checksums") or []:
```

Then it should just work in all cases, including if the behavior around appstream variants changes in the future.

Reply:
Yeah, that makes sense. Thanks!

(In case you are wondering why I did not opt to use the exact code snippet you provided,

```
>           for image in treeinfo.get("checksums") or []:
E           TypeError: get() missing 1 required positional argument: 'option'
```

It appears that a ConfigParser object's `get` method is used to obtain the value of a single key/value pair within a section, and cannot be used to get a section's collection of key/value pairs)